apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/Keys.cs

```diff
@@ -275,51 +267,49 @@ private V1.KeyRange ToProtobuf()
     /// </summary>
     public sealed class Key
     {
-        /// <summary>
-        /// The values of this key.
-        /// </summary>
-        internal ListValue KeyParts { get; }
+        internal SpannerParameterCollection Parameters { get; }
 
         /// <summary>
         /// Creates a key from the values in the parameter collection.
+        /// A reference of parameter is stored to this collection. The
+        /// changes to the collection will be visible when the key is used.
         /// </summary>
         /// <param name="keyParts">The values that define the key</param>
         internal Key(SpannerParameterCollection keyParts)
         {
-            KeyParts = ToListValue(keyParts);
+            Parameters = keyParts;
         }
 
         /// <summary>
         /// Creates a key consisting of the given key parts.
         /// </summary>
         /// <param name="keyParts">The values that define the key</param>
-        public Key(params object[] keyParts)
+        public Key(params object[] keyParts) : this(new SpannerParameterCollection(keyParts.Select(keyPart => new SpannerParameter { Value = keyPart })))
         {
-            var protobufValues = keyParts.Select(
-                v => SpannerDbType.FromClrType(v?.GetType())?.ToProtobufValue(v) ?? Value.ForNull());
-            KeyParts = new ListValue { Values = { protobufValues } };
         }
 
-        /// <inheritdoc/>
-        public override bool Equals(object other)
-            => other is Key key && KeyParts.Equals(key.KeyParts);
-
-        /// <inheritdoc/>
-        public override int GetHashCode() => KeyParts.GetHashCode();
-
-        /// <inheritdoc/>
-        public override string ToString() => KeyParts.ToString();
+        /// <summary>
+        /// Returns the string representation of this key using the type conversion options specified in 
+        /// <see cref="SpannerConnectionStringBuilder"/>. The string representation is only for diagnostic purposes
+        /// and the implementation may change over time.
+        /// </summary>
+        /// <param name="builder">The <see cref="SpannerConnectionStringBuilder"/> instance.</param>
+        /// <returns>A string that represents the current key.</returns>
+        public string ToString(SpannerConnectionStringBuilder builder) => ToProtobuf(builder.ConversionOptions).ToString();
 
-        private static ListValue ToListValue(SpannerParameterCollection parameters)
-        {
-            return new ListValue
+        /// <summary>
+        /// Gets the Protobuf value of this Key.
```

Comment:
Suggest: `Converts this key to a protobuf value, with default type conversion options as supplied in <paramref name="options" />.`